### PR TITLE
Do not run rst2ansi if a help is not printed

### PR DESCRIPTION
**Pull request by mlech-reef**

### CODE REVIEW

**Comment by mlech-reef** [comment]
20 slowest tests with the fix:

```
0.59s call     test/unit/test_console_tool.py::TestConsoleTool::test_keys
0.28s call     test/unit/test_console_tool.py::TestConsoleTool::test_copy_file_by_id
0.28s call     test/unit/test_console_tool.py::TestConsoleTool::test_files
0.27s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_bucket_complex
0.18s call     test/unit/test_console_tool.py::TestConsoleTool::test_ls
0.18s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_bucket_with_hidden
0.17s call     test/unit/test_console_tool.py::TestConsoleTool::test_buckets
0.15s call     test/unit/test_console_tool.py::TestConsoleTool::test_restrictions
0.15s call     test/unit/test_console_tool.py::TestConsoleTool::test_sync_dry_run
0.15s call     test/unit/test_console_tool.py::TestConsoleTool::test_ls_for_restricted_bucket
0.13s call     test/unit/test_console_tool.py::TestConsoleTool::test_bucket_missing_for_bucket_key
0.11s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_bucket_one_item_show_size
0.10s call     test/unit/test_console_tool.py::TestConsoleTool::test_create_key_with_authorization_from_env_vars
0.10s call     test/unit/test_console_tool.py::TestConsoleTool::test_list_buckets_not_allowed_for_app_key
0.10s call     test/unit/test_console_tool.py::TestConsoleTool::test_create_bucket_key_and_authorize_with_it
0.09s call     test/unit/test_console_tool.py::TestConsoleTool::test_cancel_large_file
0.09s call     test/unit/test_console_tool.py::TestConsoleTool::test_authorize_key_without_list_buckets
0.08s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_bucket_with_folders
0.08s call     test/unit/test_console_tool.py::TestConsoleTool::test_sync_exclude_if_modified_after_exact
0.08s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_bucket_with_versions
```

And without the fix:
```
9.67s call     test/unit/test_console_tool.py::TestConsoleTool::test_keys
3.82s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_bucket_complex
3.61s call     test/unit/test_console_tool.py::TestConsoleTool::test_copy_file_by_id
3.47s call     test/unit/test_console_tool.py::TestConsoleTool::test_files
2.86s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_bucket_with_hidden
2.32s call     test/unit/test_console_tool.py::TestConsoleTool::test_ls
2.16s call     test/unit/test_console_tool.py::TestConsoleTool::test_restrictions
2.07s call     test/unit/test_console_tool.py::TestConsoleTool::test_buckets
1.96s call     test/unit/test_console_tool.py::TestConsoleTool::test_sync_dry_run
1.77s call     test/unit/test_console_tool.py::TestConsoleTool::test_ls_for_restricted_bucket
1.72s call     test/unit/test_console_tool.py::TestConsoleTool::test_list_buckets_not_allowed_for_app_key
1.60s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_bucket_one_item_show_size
1.46s call     test/unit/test_console_tool.py::TestConsoleTool::test_bucket_missing_for_bucket_key
1.34s call     test/unit/test_console_tool.py::TestConsoleTool::test_list_parts_with_none
1.31s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_download_auth_url_with_encoding
1.29s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_bucket_with_versions
1.29s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_download_auth_url
1.28s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_download_auth_defaults
1.28s call     test/unit/test_console_tool.py::TestConsoleTool::test_list_parts_with_parts
1.27s call     test/unit/test_console_tool.py::TestConsoleTool::test_get_download_auth_explicit
```

File: b2/arg_parser.py [hunk 1]
```diff
@@ -54,24 +54,23 @@ class ArgumentParser(argparse.ArgumentParser):
...
         kwargs.setdefault('formatter_class', RawTextHelpFormatter)
         description = kwargs.get('description', None)
         if description is not None:
-            kwargs['description'] = self._format_description(description)
+            kwargs['description'] = textwrap.dedent(description) if for_docs else description
         super(ArgumentParser, self).__init__(*args, **kwargs)
 
+    def format_help(self):
+        if self.description is not None:
+            self.description = rst2ansi(self.description.encode('utf-8'))
```
Comment (mzukowski-reef) [inline b2/arg_parser.py:65]: this side-effect feels so wrong :( I know the reason, but it looks bad... and if `format_help` is run multiple times, then you will run it multiple times - is it safe? - maybe we should store information that description was formatted... then maybe we should hack everything a little bit more:
1. run `super(ArgumentParser, self).__init__(*args, **kwargs)` with "raw" description
2. add `description` property and define `setter` that sets `self._raw_description`
3. `description` getter would check for `self._formatted_description` and use either `dedent` or `rst2ansi` depending on `for_docs` value and return always `self._formatted_description`

**Review by mzukowski-reef — CHANGES_REQUESTED** [review]

File: b2/arg_parser.py (hunk 1 above)
Comment (mlech-reef) [reply b2/arg_parser.py:65]: `format_help` is never run twice in our case and probably in 99,99% of cases, so it works, but I feel you.

Comment (mlech-reef) [reply b2/arg_parser.py:65]: Running `format_help` twice has no effect

**Review by mzukowski-reef — APPROVED** [review]

**Pull request merged**